halo2_debug/src/lib.rs

```diff
@@ -1 +1,31 @@
+use rand_core::block::BlockRng;
+use rand_core::block::BlockRngCore;
+use tiny_keccak::Hasher;
+
+/// One number generator, that can be used as a deterministic Rng, outputing fixed values.
+pub struct OneNg {}
+
+impl BlockRngCore for OneNg {
+    type Item = u32;
+    type Results = [u32; 16];
+
+    fn generate(&mut self, results: &mut Self::Results) {
+        for elem in results.iter_mut() {
+            *elem = 1;
+        }
+    }
+}
+
+pub fn one_rng() -> BlockRng<OneNg> {
+    BlockRng::<OneNg>::new(OneNg {})
+}
+
```

Comment:
I have a suggestion of adding a `test_rng` like this:
```Rust
pub fn test_rng() -> ... {
    ChaCha20Rng::seed_from_u64(0xdeadbeef)
}
```

And use it in the tests instead of the `one_rng`.  The reason is that the `one_rng` is very useful for debugging (because you know what the random values are going to be), but for testing it's more realistic to use a rng that returns uniform values.  What do you think?